### PR TITLE
[#436] resource rating design (connects #436)

### PR DESCRIPTION
**Pull request by razu9861**

### CODE REVIEW

File: src/app/resources/resources.component.html [hunk 1]
```diff
@@ -67,19 +67,27 @@ <h3 class="header"><a *ngIf="!nationName" [routerLink]="['/resources/view', elem
       <mat-header-cell *matHeaderCellDef mat-sort-header i18n>Rating</mat-header-cell>
       <mat-cell *matCellDef="let element">
         <div class="list-item-rating">
-          <span class="stars">
-            <mat-icon>star</mat-icon><mat-icon>star</mat-icon><mat-icon>star</mat-icon><mat-icon>star</mat-icon><mat-icon>star</mat-icon>
-            <!-- used simple arithmetic to get 20 and converted into % -->
-            <span [ngStyle]="{'width': getRating(element.sum, element.timesRated)}">
+          <div class="stats">
+            <img src="assets/icons/male.svg" alt="male" title="male" class="male-icon"/>{{ element.maleRating === 0 ? 0 : ((element.maleRating / element.totalRating) * 100).toFixed(0) }}%
```
Comment (paulbert) [inline src/app/resources/resources.component.html]: Should be `mat-icon` with svg file added to registry

File: src/app/resources/resources.component.html [hunk 2]
```diff
@@ -67,19 +67,27 @@ <h3 class="header"><a *ngIf="!nationName" [routerLink]="['/resources/view', elem
...
+            <img src="assets/icons/woman.svg" alt="female" title="female" class="female-icon"/>{{ element.femaleRating === 0 ? 0 : ((element.femaleRating / element.totalRating) * 100).toFixed(0) }}%
```
Comment (paulbert) [inline src/app/resources/resources.component.html]: Same, should be `mat-icon`

File: src/assets/icons/male.svg [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="-260 352 90 90" style="enable-background:new -260 352 90 90;" xml:space="preserve">
+<style type="text/css">
+	.st0{fill:#118FF2;}
```
Comment (paulbert) [inline src/assets/icons/male.svg]: Remove colors from svg.  We should control color of icon with CSS

File: src/assets/icons/woman.svg [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="-49 141 512 512" style="enable-background:new -49 141 512 512;" xml:space="preserve">
+<style type="text/css">
+	.st0{fill:#FF3BC6;}
```
Comment (paulbert) [inline src/assets/icons/woman.svg]: Same, remove color

File: src/app/resources/resources.component.ts [hunk 5]
```diff
@@ -193,4 +268,64 @@ export class ResourcesComponent implements OnInit, AfterViewInit {
...
+        Object.assign(ratingData, { _id: element.ratingId, _rev: element.ratingRev });
+      }
+      this.couchService.post('ratings', ratingData)
+        .subscribe((data) => {
+           this.ngOnInit();
+        }, (error) => console.log(error));
+    }
+  }
+
+  rating(rating) {
```
Comment (paulbert) [inline src/app/resources/resources.component.ts]: This looks very similar to `postRating`

File: src/app/resources/resources.component.ts [hunk 6]
```diff
@@ -90,12 +104,82 @@ export class ResourcesComponent implements OnInit, AfterViewInit {
...
+          break;
+      }
+    }
+    if (rating.user.name) {
+      ratingInfo.hasRated = (rating.user.name === this.userService.get().name) ? rating.rate :  ratingInfo.hasRated;
+      ratingInfo.ratingId = (rating.user.name === this.userService.get().name) ? rating._id :  ratingInfo.ratingId;
+      ratingInfo.ratingRev = (rating.user.name === this.userService.get().name) ? rating._rev :  ratingInfo.ratingRev;
+      ratingInfo.comment = (rating.user.name === this.userService.get().name) ? rating.comment :  ratingInfo.comment;
+    } else {
+      // since Admin profile is not accesible
```
Comment (paulbert) [inline src/app/resources/resources.component.ts]: Admins now have a user profile, so I don't think this else section is necessary

File: src/app/resources/resources.component.ts [hunk 7]
```diff
@@ -90,12 +104,82 @@ export class ResourcesComponent implements OnInit, AfterViewInit {
...
+          femaleRating: 0,
+          hasRated: 0,
+          comment: '',
+          ratingId: '',
+          ratingRev: '',
+          userRating: (ratings[ratingIndex].user.name === this.userService.get().name) ? ratings[ratingIndex].rate : ''
+        });
+        return { ...resource, ...ratingInfo };
+      }
+      return { ...resource,  rateSum: 0, totalRating: 0, maleRating: Math.floor(Math.random() * 101), femaleRating: 0, hasRated: 0  };
```
Comment (paulbert) [inline src/app/resources/resources.component.ts]: We should replace `Math.random()` with actual data from the ratings database or initialize to 0.

**Comment by razu9861** [comment]
closing this PR, due to duplicate code, design is implemented on another issue PR no https://github.com/ole-vi/planet/pull/408

**Pull request closed without merging**